.github/workflows/check-version.yml

```
@@ -0,0 +1,87 @@
+name: Check version
+
+on:
+  push:
+    branches:
+      - develop
+      - release/*
```

I am unsure abut the triggers. It would be kind of random. Also, adding a commit to the release branch but not to the `develop` branch would cause conflicts since the `release` branch would be merged into `main` but not `develop`.

The elixir templates has a similar workflow, but it is triggered manually by the Team Lead when a new version is being worked on: https://github.com/nimblehq/elixir-templates/blob/develop/.github/workflows/bump_version.yml. That approach is more predictable and has less unexpected side effects.

---

My suggestion is to run only on the `main` branch, bump the new version, and open a pull request to `develop` 💭 

```suggestion
      - main
```

---

This is not push the new commit into the current branch, it's checking the version and create another PR to bump it when the current version is outdated. I intend to do this as I want to prevent merging the release without increasing the version.
With the above triggers, we can have 2 cases:
- When merging the first feature of a new milestone, it will check and create the bump version PR
- When creating the release PR, it will check and mark the PR failed, and create the bump version PR. We need to merge the bump version PR first, then rebase the release PR to get the latest code, including the new version